src/rlang/dyn-array.h

```diff
@@ -164,4 +134,51 @@ void r_dyn_list_poke(struct r_dyn_array* p_vec, r_ssize i, r_obj* value) {
   r_list_poke(p_vec->data, i, value);
 }
 
+static inline
+void* const * r_dyn_pop_back(struct r_dyn_array* p_arr) {
+  void* const * out = (void* const *) r_dyn_clast(p_arr);
+  --p_arr->count;
+  return out;
+}
+
+static inline
+r_ssize r__dyn_increment(struct r_dyn_array* p_arr) {
+  r_ssize loc = p_arr->count++;
+
+  if (p_arr->count > p_arr->capacity) {
+    r_ssize new_capacity = r_ssize_mult(p_arr->capacity, p_arr->growth_factor);
+    r_dyn_resize(p_arr, new_capacity);
+  }
+
+  return loc;
+}
```

Comment:
If you don't like this name, I am not tied to it. I can't think of anything better for "increment the count, resize if necessary to make sure you can actually put something there, and return the location"